### PR TITLE
:sparkles: Add Text Component

### PR DESCRIPTION
**Pull request by suman-kr**
Text component for styling the text.

**Props**:
- **variant** : 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p'
- **italicize**: boolean
- **bold**: boolean
- **fontFamily**: 'mono' | 'cursive' | 'normal' | string
- **fontSize**: number
- **padding**: string
- **margin**: string
- **color**: keyof Theme | string
- **underline**: boolean
- **strikeThrough**: boolean
- **style**: React.CSSProperties

Issue: #10

### CODE REVIEW

File: src/components/Card/Card.stories.tsx [hunk 1]
```diff
@@ -16,9 +16,12 @@ stories.add(
...
           <CardHeader subtitle='Do you Know Lorem Ipsum?'>Lorem Ipsum</CardHeader>
           <CardBody>
+            <Text>
             Lorem Ipsum is simply dummy text of the printing and
-            typesetting industry. Lorem Ipsum has been the industry's
-            standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
+            typesetting industry. <Text inline>Lorem Ipsum has been the industry's
+            standard dummy text ever since the 1500s, when an unknown printer took a galley of type and
+            scrambled it to make a type specimen book.</Text>
+            </Text>
```
Comment (anitnilay20) [inline src/components/Card/Card.stories.tsx:24]: Card should use `Text` component internally.

File: src/components/Text/Text.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  withInfo({ inline: true })(() => {
+    return (
+      <>
+        <Text variant={'h1'} bold>
+          Hello
+        </Text>
+        <Text variant={'h2'} italicize>
+          Hello
+        </Text>
+        <Text variant={'h3'} fontFamily={'cursive'} strikeThrough>
```
Comment (anitnilay20) [inline src/components/Text/Text.stories.tsx:20]:
```suggestion
        <Text variant={'h3'} fontFamily='cursive' strikeThrough>
```

Comment (anitnilay20) [reply src/components/Text/Text.stories.tsx:20]: Make this change at all the places.

File: src/components/Text/Text.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      )}
+    </ThemeConsumer>
+  );
+};
+
+export interface TextProps {
+  /**
+   *  Defines variant for heading.
+   */
+  variant?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
```
Comment (anitnilay20) [inline src/components/Text/Text.tsx:101]: We should also have `p` tag in variants.

File: src/components/Text/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export {Text} from './Text';
```
Comment (anitnilay20) [inline src/components/Text/index.ts:1]: Text should also be imported to `src/idnex.ts`.

File: src/components/Text/Text.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+   */
+  underline?: boolean;
+  /**
+   * Strikes a line through the text.
+   */
+  strikeThrough?: boolean;
+  /**
+   * Sets display property of text to inline.
+   */
+  inline?: boolean;
```
Comment (anitnilay20) [inline src/components/Text/Text.tsx:141]: Add a `style` props too for custom styling.

File: src/components/Text/Text.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    const css: React.CSSProperties = {
+      fontWeight,
+      fontStyle,
+      fontSize,
+      fontFamily,
+      padding,
+      color,
+      margin,
+      textDecoration,
+    };
```
Comment (anitnilay20) [inline src/components/Text/Text.tsx:65]: instead of using `style` use `typestyle` to generate css class.

File: src/components/Text/Text.tsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+export const Text: React.FunctionComponent<TextProps> = (props) => {
+  const display = (theme: Theme) => {
+    let fontWeight: number = 400;
+    let fontStyle = 'normal';
+    const fontFamily = props.fontFamily;
+    const fontSize = `${props.fontSize}px`;
+    const color = props.color;
+    let padding: string;
+    let margin: string;
+    let textDecoration: string;
```
Comment (anitnilay20) [inline src/components/Text/Text.tsx:21]:
```ts
let  textDecoration = '';
if (props.underline) {
  textDecoration += 'underline';
}
if (props.strikeThrough) {
  textDecoration += ' line-through'
}
```

using else looks bad.. try to implement this change else where. See where we can fit it.

File: src/components/Text/Text.tsx [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+/**
+ * Text component to style your `Text` effortlessly.
+ *
+ * **Usage**
+ * ```jsx
+ * <Text>Hello</Text>
+ * ```
+ */
+export const Text: React.FunctionComponent<TextProps> = (props) => {
+  const display = (theme: Theme) => {
```
Comment (anitnilay20) [inline src/components/Text/Text.tsx:13]: We can move this function in a helper. Also not sure why using theme..
One  thing that would be great if colors can accept theme colors too.
eg
```ts
color: Keyof Theme | string;

color: theme[props.color] || color;
```

File: src/components/Text/Text.tsx [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+    }
+    if (props.variant === 'h6') {
+      return <h6 style={css}>{props.children}</h6>;
+    }
+    return <span style={css}>{props.children}</span>;
+  };
+  return (
+    <ThemeConsumer>
+      {(theme) => (
+        <p style={{display: props.inline ? 'inline' : 'block'}}>
```
Comment (anitnilay20) [inline src/components/Text/Text.tsx:89]: don't use `P` tag here.. this should be handled by `variant`.
also avoid using style. style should be available through props for consumer to customized it.

File: src/components/Text/Text.tsx (hunk 8 above)
Comment (anitnilay20) [reply src/components/Text/Text.tsx:13]: Only variant thing should happen inside `Text` component. All the styling should stay inside helpers directory.

File: src/components/Card/Card.stories.tsx (hunk 1 above)
Comment (suman-kr) [reply src/components/Card/Card.stories.tsx]: Doesn't know how to fix this. Meanwhile, I'm removing the usage of Text inside Card Component.

File: src/components/Card/Card.stories.tsx [hunk 10]
```diff
@@ -5,7 +5,7 @@ import { withInfo } from '@storybook/addon-info';
 import { Card, CardHeader, CardBody, CardMedia, Button } from '../..';
 import { Col, Row } from '../Grid';
 import { IoMdOptions } from 'react-icons/io';
-
+import {Text} from '../Text';
```
Comment (anitnilay20) [inline src/components/Card/Card.stories.tsx:8]: We can get rid of this.

File: src/components/Text/Text.tsx [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+    if (props.variant === 'h4') {
+      return <h4 className={css}>{props.children}</h4>;
+    }
+    if (props.variant === 'h5') {
+      return <h5 className={css}>{props.children}</h5>;
+    }
+    if (props.variant === 'h6') {
+      return <h6 className={css}>{props.children}</h6>;
+    }
+    if (props.variant === 'body') {
```
Comment (anitnilay20) [inline src/components/Text/Text.tsx:53]: we should not use `body`.  `p` is good enough.

File: src/components/Text/Text.tsx [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+      return <p className={css}>{props.children}</p>;
+    }
+    return <span className={css}>{props.children}</span>;
+  };
+  return (
+    <ThemeConsumer>
+      {(theme) => (
+        <div
```
Comment (anitnilay20) [inline src/components/Text/Text.tsx:61]: why div? why  not directly return `display` function.

File: src/components/Text/Text.tsx [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+          style={{ ...props.style, display: props.inline ? 'inline' : 'block' }}
```
Comment (anitnilay20) [inline src/components/Text/Text.tsx:62]: NO styles 🤦

Comment (anitnilay20) [reply src/components/Text/Text.tsx:62]: remove display.. actually display should not  be a prop.

File: src/helpers/text.ts [hunk 14]
```diff
@@ -0,0 +1,50 @@
+import {TextProps} from '../components/Text';
+export function format(props: TextProps) {
+    let fontWeight: 'bold' | 'normal' = 'normal';
+    let fontStyle: 'normal' | 'italic' = 'normal';
+    const fontFamily = props.fontFamily;
+    let fontSize: string = '';
+    const color = props.color;
```
Comment (anitnilay20) [inline src/helpers/text.ts:7]: we need `theme` colors here too. As commented earlier.

File: src/components/Text/Text.tsx (hunk 13 above)
Comment (anitnilay20) [reply src/components/Text/Text.tsx:62]: add default props for `style`.

Comment (anitnilay20) [reply src/components/Text/Text.tsx:62]: `style={}`

File: src/helpers/text.ts [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+import { style } from 'typestyle';
+
+export function format(props: TextProps, theme: Theme) {
+
+  let fontWeight: 'bold' | 'normal' = 'normal';
+  let fontStyle: 'normal' | 'italic' = 'normal';
+  const fontFamily = props.fontFamily;
+  let fontSize: string = '';
+  let color = props.color;
+  color = theme[props.color] || color;
```
Comment (anitnilay20) [inline src/helpers/text.ts:12]: const color = theme[props.color] || color;

File: src/components/Card/CardHeader.tsx [hunk 16]
```diff
@@ -3,7 +3,6 @@ import { classes, style } from 'typestyle';
 import { Row, Col } from '../../';
 import { lightText } from '../../helpers/color';
 import { ThemeConsumer } from '../Theme/Theme';
-
```
Comment (anitnilay20) [inline src/components/Card/CardHeader.tsx:6]: Add this back.

**Pull request merged**